HungryWolfs/DetailsMeal/DetailViewController.swift

```diff
@@ -0,0 +1,156 @@
+    
+    private let viewModelDetails: DetailsViewModel = DetailsViewModel()
+    private let viewModelMeal: MealsViewModel = MealsViewModel()
+    
+    override func viewDidLoad()
+    {
+        super.viewDidLoad()
+        
+        // butoanele bifabile
+        checkButton1.setImage(UIImage(systemName: "checkmark.square"),
+                          for: [.selected])
+        checkButton2.setImage(UIImage(systemName: "checkmark.square"),
+                          for: [.selected])
+        checkButton3.setImage(UIImage(systemName: "checkmark.square"),
+                          for: [.selected])
+        
+    
+        viewModelDetails.getDetails(id: id, completion: { [weak self] in
+            self?.mealNameLabel.text = self?.viewModelDetails.details[0].name
+           
+            guard let imageURL = URL(string: (self?.viewModelDetails.details[0].thumb)!) else { return }
+            self?.mealImage.kf.setImage(with: imageURL)
+            self?.mealImage.layer.masksToBounds = true
+            self?.mealImage.layer.cornerRadius = (self?.mealImage.bounds.width)! / 2
+
+            
+            self?.measure1Label.text = self?.viewModelDetails.details[0].measure1
+            self?.measure2Label.text = self?.viewModelDetails.details[0].measure2
+            self?.measure3Label.text = self?.viewModelDetails.details[0].measure3
+            self?.ingredient1Label.text = self?.viewModelDetails.details[0].ingredient1
+            self?.ingredient2Label.text = self?.viewModelDetails.details[0].ingredient2
+            self?.ingredient3Label.text = self?.viewModelDetails.details[0].ingredient3
+            self?.mealDescription.text = self?.viewModelDetails.details[0].instructions
+            // tagurile
+            let array = self?.viewModelDetails.details[0].tags
+            let tagList = array?.components(separatedBy: ",")
+            if tagList?.count == 1
+            {
+                self?.mealTag1Label.text = tagList?[0]
+                self?.tagView1.layer.cornerRadius = 15
+                self?.tagView1.clipsToBounds = true
+                self?.tagView1.layer.borderWidth = 1
+                self?.tagView1.layer.borderColor = UIColor.red.cgColor
+            }
+            else if tagList?.count == 2
+            {
+                self?.mealTag1Label.text = tagList?[0]
+                self?.tagView1.layer.cornerRadius = 15
+                self?.tagView1.clipsToBounds = true
+                self?.tagView1.layer.borderWidth = 1
+                self?.tagView1.layer.borderColor = UIColor.red.cgColor
+                self?.mealTag2Label.text = tagList?[1]
+                self?.tagView2.layer.cornerRadius = 15
+                self?.tagView2.clipsToBounds = true
+                self?.tagView2.layer.borderWidth = 1
+                self?.tagView2.layer.borderColor = UIColor.red.cgColor
+            }
+            else if tagList?.count == 3
+            {
+                self?.mealTag1Label.text = tagList?[0]
```

Comment:
You can create separate functions that configure each tagView, and just call them from there (like: self.configureTagView1, self.configureTagView2).